cmd/tempo-serverless/handler.go

```diff
@@ -41,38 +37,53 @@ var (
 	readerOnce   sync.Once
 )
 
+type HTTPError struct {
+	Err    error
+	Status int
+}
+
 // Handler is the main entrypoint for the serverless handler. it expects a tempopb.SearchBlockRequest
 // encoded in its parameters
-func Handler(w http.ResponseWriter, r *http.Request) {
+func Handler(r *http.Request) (*tempopb.SearchResponse, *HTTPError) {
```

Comment:
Is this signature conforming to an interface we can make assertions about?

Reply:
It was an `http.HandlerFunc` b/c that's what google required. It is now a generic function that happens to work nicely for both the lambda and cloud function shims.